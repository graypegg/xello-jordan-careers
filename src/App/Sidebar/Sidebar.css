.Sidebar__wrapper--open,
.Sidebar__wrapper--closed {
  position: fixed;
  left: 0;
  top: 0;
  background: var(--dark);
  height: 100%;
  padding: 25px 30px;
  color: #FFF;
  font-weight: normal;
  font-size: 0.9rem;
  min-width: 220px;
  max-width: 290px;
  box-shadow: 0 0 20px rgba(20, 30, 20, 0.3);
  transition: transform 0.5s cubic-bezier(0.19, 1, 0.22, 1);
  z-index: 999999;
}

.Sidebar__wrapper--closed {
  transform: translateX(-100%);
}

.Sidebar__wrapper--open {
  transform: translateX(0);
}

.Sidebar__toggleButton {
  display: flex;
  justify-content: center;
  align-items: center;
  position: absolute;
  left: 100%;
  top: 105px;
  height: 50px;
  width: 50px;
  border-top-right-radius: 10px;
  border-bottom-right-radius: 10px;
  background: var(--dark);
  cursor: pointer;
}

.Sidebar__toggleButton:hover {
  background: var(--darker);
}
