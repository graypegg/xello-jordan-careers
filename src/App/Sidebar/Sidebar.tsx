import * as React from 'react'

interface ISidebarProps {
  icon: string
}

interface ISidebarState {
  open: boolean
}

class Sidebar extends React.Component<ISidebarProps, ISidebarState> {
  constructor (props: ISidebarProps) {
    super(props)

    this.state = {
      open: false
    } as ISidebarState

    this.toggleSidebar = this.toggleSidebar.bind(this)
  }

  public toggleSidebar () {
    this.setState({
      open: !this.state.open
    })
  }

  public render () {
    return (
      <aside className={this.state.open ? 'Sidebar__wrapper--open' : 'Sidebar__wrapper--closed'}>
        <div className="Sidebar__toggleButton" onClick={this.toggleSidebar}>
          <img src={this.props.icon} width="30" />
        </div>
        {this.props.children}
      </aside>
    )
  }
}

export default Sidebar