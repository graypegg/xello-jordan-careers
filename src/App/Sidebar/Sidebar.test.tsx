import * as React from 'react'
import { shallow } from 'enzyme'
import Sidebar from './Sidebar'

it('renders without crashing', () => {
  const wrapper = shallow(<Sidebar icon="abc">xyz</Sidebar>)
  expect(wrapper).toMatchSnapshot()
})

describe('toggle button', () => {
  const wrapper = shallow(<Sidebar icon="abc">xyz</Sidebar>)

  it('opens when the toggle button is clicked initially', () => {
    wrapper.find('.Sidebar__toggleButton').simulate('click')
    expect(wrapper.find('aside').hasClass('Sidebar__wrapper--open')).toBe(true)
  })

  it('closes when the toggle button is clicked again', () => {
    wrapper.find('.Sidebar__toggleButton').simulate('click')
    expect(wrapper.find('aside').hasClass('Sidebar__wrapper--closed')).toBe(true)
  })
})