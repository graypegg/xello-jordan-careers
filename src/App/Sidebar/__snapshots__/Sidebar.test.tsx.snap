// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebar
    icon="abc"
  >
    xyz
  </Sidebar>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="Sidebar__toggleButton"
          onClick={[Function]}
        >
          <img
            src="abc"
            width="30"
          />
        </div>,
        "xyz",
      ],
      "className": "Sidebar__wrapper--closed",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            src="abc"
            width="30"
          />,
          "className": "Sidebar__toggleButton",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "abc",
            "width": "30",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      "xyz",
    ],
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="Sidebar__toggleButton"
            onClick={[Function]}
          >
            <img
              src="abc"
              width="30"
            />
          </div>,
          "xyz",
        ],
        "className": "Sidebar__wrapper--closed",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              src="abc"
              width="30"
            />,
            "className": "Sidebar__toggleButton",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": "abc",
              "width": "30",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        "xyz",
      ],
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
